rust/chains/hyperlane-cosmos/src/mailbox.rs

```diff
@@ -352,10 +354,32 @@ impl Indexer<HyperlaneMessage> for CosmosMailboxIndexer {
         &self,
         range: RangeInclusive<u32>,
     ) -> ChainResult<Vec<(HyperlaneMessage, LogMeta)>> {
-        let result = self
-            .indexer
-            .get_range_event_logs(range, Self::hyperlane_message_parser)
-            .await?;
+        let logs_futures: Vec<_> = range
+            .map(|block_number| {
+                let self_clone = self.clone();
+                tokio::spawn(async move {
+                    self_clone
+                        .indexer
+                        .get_event_log(block_number, Self::hyperlane_message_parser)
+                        .await
+                })
+            })
+            .collect();
+
+        // TODO: this can be refactored when we rework indexing, to be part of the block-by-block indexing
+        let result = future::join_all(logs_futures)
+            .await
+            .into_iter()
+            .flatten()
+            .filter_map(|res| match res {
+                Ok(logs) => Some(logs),
+                Err(err) => {
+                    warn!(?err, "Failed to fetch logs");
```

Comment:
the nice part here is for the message & merkle tree indexing we'll automatically rewind to try these again, so less concerned here

Reply:
Should I remove the warning?

Reply:
No I think the warning is useful, can you also add the block num here for consistency with the IGP indexing?